### PR TITLE
Fix `Tribe__Admin__Helpers::is_screen()` logic

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://central.tri.be/issues/107413

To avoid false positive and return the correct result (i.e.: On the events tag page, and returning true on many non-aggregator pages as aggregator pages).

### CODE REVIEW

File: src/Tribe/Admin/Helpers.php [hunk 1]
```diff
@@ -109,14 +109,14 @@ public function is_screen( $id = null ) {
...
-			return true;
+		if ( ! is_null( $id ) && $current_screen->id !== $id ) {
+			return false;
 		}
 
 		// Match any post type page in the supported post types
 		$defaults = apply_filters( 'tribe_is_post_type_screen_post_types', Tribe__Main::get_post_types() );
-		if ( ! in_array( $current_screen->post_type, $defaults ) ) {
-			return false;
+		if ( in_array( $current_screen->post_type, $defaults ) ) {
```
Comment (mitogh) [inline src/Tribe/Admin/Helpers.php]: Seems like you can simplify this section of code by removing the next two returns and just returning the result of `in_array` as will return either `true/false` depending on your operation.

```php
return in_array( $current_screen->post_type, $defaults );
```

**Comment by juanfra** [comment]
@bordoni I've been doing dev QA and everything seems to be working as it should. I also flagged that you requested to have QA in branch for this ticket during the scrum.

What type of tests do you feel this ticket requires?

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *M18.16*, but this was pull requested against *release/M18.15*.

**Comment by ggwicz** [comment]
This has been moved to the _Prioritized Backlog_, so I've changed its base branch from a specific release branch to `master`.

⚠️  **Please do not delete or further adjust this PR** without pinging me or @juanfra here or on Slack.

**Comment by juanfra** [comment]
Closing in favor of https://github.com/moderntribe/tribe-common/pull/783/

**Pull request closed without merging**